### PR TITLE
Switch to storing sessions in db

### PR DESCRIPTION
**Pull request by amccloud**

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -1,48 +1,53 @@
...
 }
 
 app.use(express.static(contentBase));
+app.use(session({
+  secret: process.env.AUTH_SECRET,
+  store: new SessionStore({db}),
+  saveUninitialized: false,
+  resave: false,
+  proxy: true
+}));
```
Comment (zpnk) [inline src/server.js]: By default express-sessions does not set a maxAge or expiry on the cookie. Consequently, the browser interprets these cookies as "non-persistant" meaning they are cleared on browser close.  [[source](https://github.com/expressjs/session#cookieexpires)] This is reproducible by logging in then restarting the browser.

I think we should set `cookie.maxAge` to one year (~`1000 * 60 * 60 * 24 * 365`). Additionally, I think we'll want to set [`cookie.secure`](https://github.com/expressjs/session#cookiesecure) to 'auto'.

Also the default cookie name is 'connect.sid'. I'd like to rename this to something like 'cp_token' or 'cp_session'. `name: 'cp_token'`

File: src/server/authApi.js [hunk 2]
```diff
@@ -74,7 +74,14 @@ router.post('/token', async (request, response) => {
...
   }
 
-  response.send({token: tokenForUser(user)});
+  setUser(request, user, (user) => response.send(user));
+});
+
+router.delete('/session', async (request, response) => {
+  request.session.destroy((error) => {
+    if (error) { throw error; }
+    response.send({session: {}});
```
Comment (zpnk) [inline src/server/authApi.js]: If there is an error here, this `throw` will cause the node process to crash. Is this desired?

**Comment by amccloud** [comment]
I'm not expecting there to ever be an error but when it happens It will be
caught by Sentry and the process should continue.

File: src/server/authApi.js (hunk 2 above)
Comment (amccloud) [inline src/server/authApi.js]: Sentry will catch the error

**Pull request merged**